core/variant_op.cpp

```diff
@@ -322,8 +336,8 @@ void Variant::evaluate(const Operator& p_op, const Variant& p_a, const Variant&
 				DEFAULT_OP_ARRAY_EQ(POOL_INT_ARRAY,int);
 				DEFAULT_OP_ARRAY_EQ(POOL_REAL_ARRAY,real_t);
 				DEFAULT_OP_ARRAY_EQ(POOL_STRING_ARRAY,String);
-				DEFAULT_OP_ARRAY_EQ(POOL_VECTOR2_ARRAY,Vector3);
-				DEFAULT_OP_ARRAY_EQ(POOL_VECTOR3_ARRAY,Vector3);
+				DEFAULT_OP_ARRAY_NANEQ(POOL_VECTOR2_ARRAY,Vector3);
+				DEFAULT_OP_ARRAY_NANEQ(POOL_VECTOR3_ARRAY,Vector3);
```

Comment:
Not completely sure what you are doing here, but shouldn't you add a `DEFAULT_OP_ARRAY_NANEQ` for `POOL_REAL_ARRAY` (and maybe Int array)?